wasm/src/decoder/code_section.rs

```diff
@@ -0,0 +1,32 @@
+use super::common::CodeBlock;
+use super::util;
+use crate::error::Result;
+
+pub fn decode_code_section<R: crate::Reader>(reader: &mut R) -> Result<CodeBlock> {
+    let body_size = util::decode_u32(reader)? as u64;
+    let offset = reader.seek(std::io::SeekFrom::Current(0))?;
+
+    let v = util::decode_vec(reader, |x| {
+        Ok((util::decode_u32(x)?, util::decode_kind(x)?))
+    })?;
+
+    let mut locals = Vec::new();
+    for x in v {
+        for _ in 0..x.0 {
+            locals.push(x.1);
+        }
```

Comment:
Hmm, this could extend via an interior?
`locals.extend((0..x.0).map(|_| x.1)` or maybe `locals.resize(locals.len() + x.0, x.1)` or maybe this is overkill anyways

Reply:
Was trying that. But I wasn't able to figure it out. Gonna change that

Reply:
I'll do it in a follow-up PR

Reply:
You should be able to just push to this pr branch

Reply:
I'll save it for an optimisation PR when we have tests and benchmarking.